src/definitions/modules/dropdown.js

```diff
@@ -2882,6 +2882,14 @@ $.fn.dropdown = function(parameters) {
               module.debug('Cannot select blank values from multiselect');
               return;
             }
+
+            if(settings.fireOnInit === false && module.is.initialLoad()) {
+              module.verbose('Skipping onbeforeadd callback on initial load', settings.onBeforeAdd);
+            }
+            else {
```

Comment:
I checked the code so far. Basically the `module.add.value` method is the right place, but you have to adjust the check for labels here
https://github.com/fomantic/Fomantic-UI/blob/4b5cd6bc0dd17cf2645d3a5f1564cfe20691b7a3/src/definitions/modules/dropdown.js#L2680-L2692

The `module.add.value(selectedValue, selectedText, $selected); ` should be called in any case and all the followup instructions should be wrapped into an if statement depending on the return value (as i mentioned before: If onBeforeAdd returns false, the whole adding of a value should be permitted

I would rewrite this as

```javascript
if(module.add.value(selectedValue, selectedText, $selected) !== false && settings.useLabels) { 
     module.add.label(selectedValue, selectedText, shouldAnimate); 
     module.set.activeItem($selected); 
     module.filterActive(); 
     module.select.nextAvailable($selectedItem); 
   } 
   else { 
     module.set.text(module.add.variables(message.count)); 
     module.set.activeItem($selected); 
   } 
```

Reply:
@lubber-de 
> as i mentioned before: If onBeforeAdd returns false, the whole adding of a value should be permitted
wait, you said 
> if onBeforeAdd returns false to prevent adding the value
did I misunderstand? should it stop adding or _not_ stop adding the value, if onBeforeAdd returns false?

Reply:
If `onBeforeAdd` returns false it should not add the value. This would replicate the behavior of similar events in the framework.

Reply:
Oh, yes, @hammy2899 is right, i used the wrong words 🙄 I should have written "prevented" instead of "permitted"